crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -814,6 +814,47 @@ pub(super) fn concat_hor(
     delimiter: &str,
     ignore_nulls: bool,
 ) -> PolarsResult<Series> {
+    // If delimiter is empty and ignore_nulls is false, we'll check for nulls in the data
+    if delimiter.is_empty() && !ignore_nulls {
+        // convert to string arrays for null checking
+        let str_series: Vec<_> = series
+            .iter()
+            .map(|s| match s.dtype() {
+                DataType::List(_) => s
+                    .cast(&DataType::List(Box::new(DataType::String))),
+                _ => s.cast(&DataType::String),
+            })
+            .collect::<PolarsResult<_>>()?;
+
+        // Check if any array has nulls, if so, return null array
+        let has_nulls = str_series.iter().any(|s| match s.dtype() {
```

Comment:
Are you sure this is the correct logic?
You are checking if there is a null in the entire series, each row should be considered individually

Reply:
Also, please test with a more diverse dataset, what happens when concating both list[str] and str, what happens if the list[str] is null? What happens if only one of the elements is null?